### PR TITLE
Feat/entity/currency

### PR DESCRIPTION
**Pull request by essteer**
Adds a `Currency` class with two attributes:

- Primary key of `currencyCode` - validation in place to ensure it is a three-letter String
- `currencyName` - validation in place to ensure it is not null

Unit tests added and passing.

### CODE REVIEW

File: src/main/java/com/fdmgroup/forex/services/CurrencyService.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        this.currencyRepo = currencyRepo;
+    }
+
+    public Currency findById(String id) throws RecordNotFoundException {
+        Optional<Currency> currencyOptional = currencyRepo.findById(id);
+        return currencyOptional.orElseThrow(() -> 
+            new RecordNotFoundException("Currency with id '" + id + "'' not found"));
+    }
+
+    public Currency findByCurrencyName(String currencyName) throws RecordNotFoundException {
```
Comment (dexterklui) [inline src/main/java/com/fdmgroup/forex/services/CurrencyService.java]: I don't think we will ever need this service?

File: src/main/java/com/fdmgroup/forex/services/CurrencyService.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.springframework.stereotype.Service;
+
+import com.fdmgroup.forex.exceptions.RecordNotFoundException;
+import com.fdmgroup.forex.models.Currency;
+import com.fdmgroup.forex.repos.CurrencyRepo;
+
+@Service
+public class CurrencyService {
+
```
Comment (dexterklui) [inline src/main/java/com/fdmgroup/forex/services/CurrencyService.java:14]: I think what is needed the most is `findAllCurrencies` method?
I imagine frontend will need to fetch all currencies right in the beginning and store it for later use:
e.g. when displaying each portfolio asset, frontend will need to use its stored currencies info and show the current currency name.

**Comment by essteer** [comment]
@dexterklui Noted, thanks - I've removed the findByCurrencyName method and added findAllCurrencies. Mappings have been added to the controller to get all currencies and get a single currency by ID.

File: src/main/java/com/fdmgroup/forex/models/Currency.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import jakarta.persistence.Id;
+
+@Entity
+public class Currency {
+
+    @Id
+    private String currencyCode;
+
+    @Column(unique = true, nullable = false)
+    private String currencyName;
```
Comment (dexterklui) [inline src/main/java/com/fdmgroup/forex/models/Currency.java:16]: Add `@Pattern` validation

File: src/main/java/com/fdmgroup/forex/models/Currency.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        return currencyName;
+    }
+
+    public void setCurrencyName(String newName) {
+        if (newName != null && newName != "") {
+            this.currencyName = newName;
+        }
+    }
+
+    public String toString() {
```
Comment (dexterklui) [inline src/main/java/com/fdmgroup/forex/models/Currency.java:46]: Add `@Override`

File: src/main/java/com/fdmgroup/forex/models/Currency.java (hunk 3 above)
Comment (essteer) [reply src/main/java/com/fdmgroup/forex/models/Currency.java:16]: Resolved in 6ab42a9

File: src/main/java/com/fdmgroup/forex/models/Currency.java (hunk 4 above)
Comment (essteer) [reply src/main/java/com/fdmgroup/forex/models/Currency.java:46]: Resolved in 1b92d44

**Pull request merged**